storage/src/journal/fixed.rs

```diff
@@ -0,0 +1,656 @@
+            Err(err) => return Err(Error::Runtime(err)),
+        };
+        for name in stored_blobs {
+            let blob = runtime
+                .open(&cfg.partition, &name)
+                .await
+                .map_err(Error::Runtime)?;
+            let index = match name.try_into() {
+                Ok(index) => u64::from_be_bytes(index),
+                Err(nm) => return Err(Error::InvalidBlobName(hex(&nm))),
+            };
+            debug!(blob = index, "loaded blob");
+            blobs.push((blob, index));
+        }
+        blobs.sort_by(|(_, a), (_, b)| a.cmp(b));
+        let mut oldest_blob = 0;
+        if !blobs.is_empty() {
+            // Check that there are no gaps in the blob numbering, which would indicate missing data.
+            let mut it = blobs.iter();
+            let mut previous_blob = it.next().unwrap().1;
+            oldest_blob = previous_blob;
+            for (_, section) in it {
+                if *section != previous_blob + 1 {
+                    return Err(Error::MissingBlob(previous_blob + 1));
+                }
+                previous_blob = *section;
+            }
+        } else {
+            debug!("no blobs found, bootstrapping with 1 empty blob");
+            let blob = runtime.open(&cfg.partition, &0u64.to_be_bytes()).await?;
+            blobs.push((blob, 0));
+        }
+
+        // Initialize metrics
+        let tracked = Gauge::default();
+        let synced = Counter::default();
+        let pruned = Counter::default();
+        {
+            let mut registry = cfg.registry.lock().unwrap();
+            registry.register("tracked", "Number of blobs", tracked.clone());
+            registry.register("synced", "Number of syncs", synced.clone());
+            registry.register("pruned", "Number of blobs pruned", pruned.clone());
+        }
+        tracked.set(blobs.len() as i64);
+
+        let blobs: Vec<B> = blobs.into_iter().map(|(blob, _)| blob).collect();
+        // truncate the last blob if it's not the expected length, which might happen from unclean shutdown.
+        let last_blob = blobs.last().unwrap();
+        let blob_len: u64 = last_blob.len().await?;
+        if blob_len % (size_of::<u32>() + N) as u64 != 0 {
+            warn!(
+                "last blob len ({}) is not a multiple of item size, truncating",
+                blob_len
+            );
+            last_blob
+                .truncate(blob_len - blob_len % (size_of::<u32>() + N) as u64)
+                .await?;
+            last_blob.sync().await?;
+        }
+
+        Ok(Self {
+            runtime,
+            cfg,
+            oldest_unpruned_blob: oldest_blob as usize,
+            blobs,
+            tracked,
+            synced,
+            pruned,
+        })
+    }
+
+    /// Sync any pending updates to disk.
+    pub async fn sync(&mut self) -> Result<(), Error> {
+        self.synced.inc();
+        debug!("syncing journal {:?}", self.synced);
```

Comment:
`synced` won't necessarily return the blob you are syncing. `sync` may be called before the end of a "run" so `self.synced` could be some random value. 

I think you need to get the value by parsing `self.blobs`?

Reply:
that should have been self.most_recent_blob_number().  fixed.